src/wasm/chaining_util.cpp

```diff
@@ -4,6 +4,13 @@
 #include <faabric/scheduler/Scheduler.h>
 #include <faabric/util/bytes.h>
 
+//TODO the logic for sgx chain calls should probably not live here, but in src/sgx?
+//TODO heavy code duplication, this needs refactoring!
```

Comment:
This isn't something that should be left in a `TODO`, it'll need to be done before we merge the PR